app/controllers/tags_controller.rb

```diff
@@ -176,31 +176,36 @@ def new
   # POST /tags
   def create
     type = tag_params[:type] if params[:tag]
-    if type
-      raise "Redshirt: Attempted to constantize invalid class initialize create #{type.classify}" unless Tag::TYPES.include?(type.classify)
-      model = begin
-                type.classify.constantize
-              rescue
-                nil
-              end
-      @tag = model.find_or_create_by_name(tag_params[:name]) if model.is_a? Class
-    else
-      flash[:error] = ts('Please provide a category.')
+
+    unless type
+      flash[:error] = ts("Please provide a category.")
       @tag = Tag.new(name: tag_params[:name])
-      render(action: 'new') && return
+      render(action: "new")
+      return
     end
-    if @tag && @tag.valid?
-      if (@tag.name != tag_params[:name]) && @tag.name.casecmp(tag_params[:name].downcase).zero? # only capitalization different
-        @tag.update_attribute(:name, tag_params[:name]) # use the new capitalization
-        flash[:notice] = ts('Tag was successfully modified.')
-      else
-        flash[:notice] = ts('Tag was successfully created.')
-      end
+
+    raise "Redshirt: Attempted to constantize invalid class initialize create #{type.classify}" unless Tag::TYPES.include?(type.classify)
+
+    model = begin
+              type.classify.constantize
+            rescue StandardError
+              nil
+            end
+    @tag = model.find_or_create_by_name(tag_params[:name]) if model.is_a? Class
+
+    unless @tag&.valid?
+      render(action: "new")
+      return
+    end
+
+    if @tag.id_previously_changed? # i.e. tag is new
       @tag.update_attribute(:canonical, tag_params[:canonical])
-      redirect_to edit_tag_path(@tag)
+      flash[:notice] = ts("Tag was successfully created.")
     else
-      render(action: 'new') && return
+      flash[:notice] = ts("Tag already existed and was not modified.")
     end
```

Comment:
The only real change is within this block, that replaces this one:
https://github.com/otwcode/otwarchive/blob/master/app/controllers/tags_controller.rb#L192..L201

The rest is a small syntax change* to keep the number of imbricated if low, and accidentally awakening the hound while doing so.

*Replacing:
```
if condition
  [normal behavior]
else
  [error handling]
  return
end
```
With:
```
unless condition
  [error handling]
  return
end

[normal behavior]
```

Reply:
Yeah that's totally fine.

FYI when linking to code, you can [get a link to specific commits](https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files) by pressing Y.